compiler/rustc_mir/src/borrow_check/mod.rs

```diff
@@ -1374,13 +1359,29 @@ impl<'cx, 'tcx> MirBorrowckCtxt<'cx, 'tcx> {
 
     fn propagate_closure_used_mut_upvar(&mut self, operand: &Operand<'tcx>) {
         let propagate_closure_used_mut_place = |this: &mut Self, place: Place<'tcx>| {
-            if !place.projection.is_empty() {
-                if let Some(field) = this.is_upvar_field_projection(place.as_ref()) {
+            // In case of nested closure, what we might be mutating is
+            // a. Is a captured place in our parent closure,
+            // b. Is some path starting from a captured place in our parent closure
+            // c. Is a path starting from a local variable in our parent closure.
+
+            // Handle (a)
```

Comment:
I think this is maybe missing a case -- if we capture a deref of an `&mut`. Also, I don't think that `(a)` is distinct from `(b)`.